charts/federation-v2/README.md

```diff
@@ -83,13 +83,17 @@ chart and their default values.
 
 | Parameter | Description | Default |
 |-----------|-------------|---------|
-| controllermanager.replicaCount | Number of replica for federation v2 controller manager | 1 |
-| controllermanager.repository | Repo of the federation v2 image | quay.io/kubernetes-multicluster |
-| controllermanager.image | Name of the federation v2 image | federation-v2 |
-| controllermanager.tag | Tag of the federation v2 image | latest |
-| controllermanager.pullPolicy | Image pull policy | IfNotPresent |
-| clusterregistry.enabled | Specifies whether to enable the clusterregistry in federation v2 | true |
-| federatedtypes.enabled | Specifies whether to install the `FederatedTypeConfig` in federation v2 | true |
+| controllermanager.replicaCount | Number of replica for federation v2 controller manager. | 1 |
+| controllermanager.repository | Repo of the federation v2 image. | quay.io/kubernetes-multicluster |
+| controllermanager.image | Name of the federation v2 image. | federation-v2 |
+| controllermanager.tag | Tag of the federation v2 image. | latest |
+| controllermanager.pullPolicy | Image pull policy. | IfNotPresent |
+| controllermanager.featureGates | Feature gates are a set of `key=value` pairs that describe alpha or experimental features. An administrator can use the `--feature-gates` command line flag on each component to turn a feature on or off. | PushReconciler=true,SchedulerPreferences=true,CrossClusterServiceDiscovery=true,FederatedIngress=true |
+| controllermanager.fedNamespace | The namespace the federation control plane is deployed in. | federation-system |
+| controllermanager.clusterNamespace | The cluster registry namespace. | kube-multicluster-public |
+| controllermanager.clusterNamespace | Whether the federation namespace will be the only target for federation. | false |
+| clusterregistry.enabled | Specifies whether to enable the clusterregistry in federation v2. | true |
```

Comment:
What does this flag imply?  Since 'cluster registry' is primarily about installing the CRD, can't this be done implicitly (i.e. if crd isn't present, install it)?  Maybe the only flag should be for creating the `kube-multicluster-public` namespace?

Reply:
If the CRD is installed, great.  If it is not installed, it needs to be installed.  I think this behavior should be automatic, and not controlled by a flag.

For a cluster-scoped federation deployment, `kube-multicluster-public` is the default registry namespace and federation will fail to work if it does not exist.  I think this implies that the chart should create it.

Reply:
@marun If the CRD already exist, and when the helm chart include the CRD and if install the CRD again, the `helm install` will be failed, so seems we need a flag to control this? How to make this behavior automatic?

I will try to see if I can create the namespace in helm chart, but how about handle the namespace create in helm chart in another PR?

Reply:
I'm hoping helm is capable enough to be able to check for the CRD and install if needed rather than forcing the user to perform that mechanical action.  Is that not the case?

I'm fine with the namespace create in a separate PR.

Reply:
After some check, I think helm do not have such capability now, so here I was using some condition checking. But I will check more for this in helm community.